esphome/components/mitsubishi/mitsubishi.cpp

```diff
@@ -22,37 +35,129 @@ const uint16_t MITSUBISHI_HEADER_MARK = 3500;
 const uint16_t MITSUBISHI_HEADER_SPACE = 1700;
 const uint16_t MITSUBISHI_MIN_GAP = 17500;
 
+// Marker bytes
+const uint8_t MITSUBISHI_BYTE00 = 0X23;
+const uint8_t MITSUBISHI_BYTE01 = 0XCB;
+const uint8_t MITSUBISHI_BYTE02 = 0X26;
+const uint8_t MITSUBISHI_BYTE03 = 0X01;
+const uint8_t MITSUBISHI_BYTE04 = 0X00;
+const uint8_t MITSUBISHI_BYTE13 = 0X00;
+const uint8_t MITSUBISHI_BYTE16 = 0X00;
+
 void MitsubishiClimate::transmit_state() {
-  uint32_t remote_state[18] = {0x23, 0xCB, 0x26, 0x01, 0x00, 0x20, 0x08, 0x00, 0x30,
-                               0x58, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00};
+  // Byte 0-4: Constant: 0x23, 0xCB, 0x26, 0x01, 0x00
+  // Byte 5: On=0x20, Off: 0x00
+  // Byte 6: MODE (See MODEs above (Heat/Dry/Cool/Auto)
+  // Byte 7: TEMP bits 0,1,2,3, added to MITSUBISHI_TEMP_MIN
+  //          Example: 0x00 = 0°C+MITSUBISHI_TEMP_MIN = 16°C; 0x07 = 7°C+MITSUBISHI_TEMP_MIN = 23°C
+  // Byte 8: MODE_A & Wide Vane (if present)
+  //          MODE_A bits 0,1,2 different than Byte 6 (See MODE_As above)
+  //          Wide Vane bits 4,5,6,7 (Middle = 0x30)
+  // Byte 9: FAN/Vertical Vane/Switch To Auto
+  //          FAN (Speed) bits 0,1,2
+  //          Vertical Vane bits 3,4,5 (Auto = 0x00)
+  //          Switch To Auto bits 6,7
+  // Byte 10: CLOCK Current time as configured on remote (0x00=Not used)
+  // Byte 11: END CLOCK Stop time of HVAC (0x00 for no setting)
+  // Byte 12: START CLOCK Start time of HVAC (0x00 for no setting)
+  // Byte 13: Constant 0x00
+  // Byte 14: HVAC specfic, i.e. ECONO COOL, CLEAN MODE, always 0x00
+  // Byte 15: HVAC specfic, i.e. POWERFUL, SMART SET, PLASMA, always 0x00
+  // Byte 16: Constant 0x00
+  // Byte 17: Checksum: SUM[Byte0...Byte16]
+
+  uint32_t remote_state[18] = {0x23, 0xCB, 0x26, 0x01, 0x00, 0x20, 0x08, 0x00, 0x00,
+                               0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00};
 
   switch (this->mode) {
-    case climate::CLIMATE_MODE_COOL:
-      remote_state[6] = MITSUBISHI_COOL;
-      break;
     case climate::CLIMATE_MODE_HEAT:
-      remote_state[6] = MITSUBISHI_HEAT;
+      remote_state[6] = MITSUBISHI_MODE_HEAT;
+      remote_state[8] = MITSUBISHI_MODE_A_HEAT;
+      break;
+    case climate::CLIMATE_MODE_DRY:
+      remote_state[6] = MITSUBISHI_MODE_DRY;
+      remote_state[8] = MITSUBISHI_MODE_A_DRY;
       break;
+    case climate::CLIMATE_MODE_COOL:
+      remote_state[6] = MITSUBISHI_MODE_COOL;
+      remote_state[8] = MITSUBISHI_MODE_A_COOL;
```

Comment:
I've been hitting a bug where setting it to COOL sets it to AUTO instead - here's why: missing a `break;`
```suggestion
      remote_state[8] = MITSUBISHI_MODE_A_COOL;
      break;
```

Reply:
Thanks for finding this. The PR has been updated.